pkg/csi_deployment.go

```diff
@@ -21,62 +22,51 @@ type CSIDeployment struct {
 	nodeController NodeController
 }
 
-func NewCSIDeployment(ctx context.Context, clientSet kubernetes.Clientset,
-	client client.Client, log logr.Logger) CSIDeployment {
+func NewCSIDeployment(clientSet kubernetes.Clientset, client client.Client, log logr.Logger) CSIDeployment {
 	return CSIDeployment{
 		node: node.NewNode(
-			ctx,
 			&clientSet,
 			log.WithValues(constant.CSIName, "node"),
 		),
 		controller: Controller{
-			ctx:       ctx,
 			Clientset: clientSet,
 			Logger:    log.WithValues(constant.CSIName, "controller"),
 		},
 		extender: SchedulerExtender{
-			ctx:       ctx,
 			Clientset: clientSet,
 			Logger:    log.WithValues(constant.CSIName, "extender"),
 		},
 		patcher: SchedulerPatcher{
-			ctx:       ctx,
 			Clientset: clientSet,
 			Client:    client,
 			Logger:    log.WithValues(constant.CSIName, "patcher"),
 		},
 		nodeController: NodeController{
-			ctx:       ctx,
 			Clientset: clientSet,
 			Logger:    log.WithValues(constant.CSIName, "nodeController"),
 		},
 	}
 }
 
 func (c *CSIDeployment) Update(ctx context.Context, csi *csibaremetalv1.Deployment, scheme *runtime.Scheme) error {
-	if err := c.node.Update(csi, scheme); err != nil {
-		return err
-	}
-
-	if err := c.controller.Update(csi, scheme); err != nil {
-		return err
-	}
+	errs, ctx := errgroup.WithContext(ctx)
 
-	if err := c.extender.Update(csi, scheme); err != nil {
-		return err
-	}
+	errs.Go(func() error { return c.nodeController.Update(ctx, csi, scheme) })
```

Comment:
Why do you want to deploy components in parallel here? Shouldn't Patcher be run when Extender is deployed only?

Reply:
Currently, if we have an error in the stage, following stages after failed one are not started at all. It makes slower deploying process and increases API calls number. 
Extender will wait until Patcher have done its work in binding stage. And if we want to start it after patch, we need to check if Patcher is running and kube-scheduler is restarted. At the moment we creates them only and don't do these checks

Reply:
I guess this is expected behavior when one component failed to install - don't proceed with others. What do you mean by increased number of API calls?

Reply:
I mean when there is an error while reconciling (updating here), we don't really process all components only ones before an error and if one more error below appears, controller has to retry reconcile one more time. As for creating, our components are independent from each other, they can wait before others have been ready after deploying.

But I agree that it's better not to try creating new components if one was failed. What order do you propose? Patcher -> Extender -> NodeController -> Controller -> Node?

Reply:
NodeController -> Driver -> Extender/Patcher

Reply:
Done